### ISSUE TITLE
Femtovg renderer selection glitches with ligatures

### ISSUE THREAD

**Issue by ogoffart** [slint-ui/slint#2968]
To reproduce: in the gallery, with the femtovg renderer, go to an edit field and type string with ligatures (for example something with fi or ffi or something like that), and use the keyboard (with shift-arrow) to select text, and observe glitches in the selection.

[demo](https://slint.dev/snapshots/master/editor/?snippet=import+%7B+Button%2C+VerticalBox+%2C+LineEdit%2C+TextEdit%7D+from+%22std-widgets.slint%22%3B%0Aexport+Demo+%3A%3D+Window+%7B%0A++++default-font-size%3A+20px%3B%0A++++width%3A+200px%3B+height%3A+200px%3B%0A++++VerticalBox+%7B%0A++++++++TextEdit+%7B+%0A++++++++++++text%3A+%22ffiiifffifiifffi+iiiiffififffiifiiiifff+fffiiifiififiiiffifififiiifii+fifi%22%3B%0A++++++++%7D%0A%0A++++++++TextInput+%7B%0A++++++++++++text%3A+%22ffiiifffifiifffi+iiiiffififffi%5Cnifiiiifff+fffiiifiififii%5Cniffifififiiifii+fifi%22%3B%0A++++++++++++single-line%3A+false%3B%0A++++++++++++cursor-position-byte-offset%3A+5%3B%0A++++++++++++anchor-position-byte-offset%3A+42%3B%0A++++++++%7D%0A++++%7D%0A%7D%0A)

<img width="150" alt="image" src="https://github.com/slint-ui/slint/assets/959326/0c112451-323c-4e30-9051-c3e8b77c95d1">

**Comment by R-Cramer4** [comment]
I can no longer reproduce this with the most recent slint version. (winit-femtovg on https://github.com/slint-ui/slint/commit/2bb5a20694e75d2e8d50cbea91595f8ebff0d9a2 and MacOS 14.6.1)

**Comment by tronical** [comment]
That's right, commit a60f54236f27523ae7d9b9e7ad8dd4a6499ca6a4 fixed this.
